### ISSUE TITLE
Test using MySQL

### ISSUE THREAD

**Issue by jhawthorn** [solidusio/solidus#469]
Our CircleCI doesn't test MySQL. Either it should, or we should use travis.

**Comment by mbj** [comment]
CircleCI is to my experience the better CI. But just lacks the matrix.

As I said in #466 I think we can expand the `build-ci.rb` script we currently use to trigger each project also under a different `DB` environment variable. Simulating the matrix feature.

Additional passes could be made to report each sub-project/DB variable combination build in separate via the github API.

If I get work authorized by the client, I could take this one.

**Comment by jhawthorn** [comment]
> CircleCI is to my experience the better CI. But just lacks the matrix.

I do not share this opinion at all, but the rest of the team seems to like it.

**Comment by mbj** [comment]
> I do not share this opinion at all, but the rest of the team seems to like it.

Curious. Can you expand your thoughts. As I do not share it (right now) there might be some news in your view.

**Comment by jhawthorn** [comment]
First, I prefer using travis (UI, configuration format). I've found it to have better uptime (anecdotal, but Circle seems to have some outage almost every week). CircleCI is appallingly bad at permissions (though this doesn't matter in an open source project). There's also no matrix builds on Circle (which is really what we're looking for right now).

I like travis because it doesn't seem to have the same 4-container limit, it runs whatever it has resources to do, but it feels marginally slower per-container. If you haven't used it in the past ~year, it's much faster to start testing now if you specify `sudo: false` in your travis.yml, which makes it use containers instead of a VM.

**Comment by mbj** [comment]
> I like travis because it doesn't seem to have the same 4-container limit,

I run with much more containers at work. I think you mean the 4 container "free" limit? A project at the intentions / goals / business-value of Solidus should not have to use the OSS tier.

**Comment by tvdeyen** [comment]
In the past Travis supported large OpenSource Projects like Rails and Spree with more free resources.
We could ask them to support Solidus

**Comment by mbj** [comment]
I'm fine with any solution as long the bottleneck is solved.

**Comment by athal7** [comment]
I've personally had troubles with Travis' stability, but not opposed to giving it another shot.

**Comment by mbj** [comment]
> I've personally had troubles with Travis' stability, but not opposed to giving it another shot.

same. Significant problems. Especially determinism. Lots of VM noise killing timing measurements. In consistent host envs etc all introducing entropy to create non-repeatable builds.

Also the UI was utterly slow the moment I last time tried Travis. And yeah, I'd also be happy to try it out.

**Comment by BenMorganIO** [comment]
My 2 cents: Switch to Travis since it provides a matrix. I've always seen Travis as the go to open source solution and CircleCI as the private solution...

**Comment by jhawthorn** [comment]
I've enabled travis for branches (not PRs yet) we can evaluate it over the next week to see if it runs "stable" enough to enable for PRs.

FWIW, I blieve that whatever instability travis may introduce can be fixed by using capybara properly.

**Comment by BenMorganIO** [comment]
@jhawthorn :+1:

**Comment by tvdeyen** [comment]
> FWIW, I believe that whatever instability travis may introduce can be fixed by using capybara properly.

Yes, Feature specs on Travis CI makes you nuts sometime. But they are handable.
